lib/shelley/bench/Restore.hs

```diff
@@ -223,86 +222,70 @@ cardanoRestoreBench tr c socketFile = do
     sayErr $ "Network: " <> network
 
     prepareNode (trMessageText tr) networkProxy socketFile np vData
-    runBenchmarks
-        [ bench_restoration @_ @ShelleyKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletSeq "0-percent-seq" $ mkSeqAnyState' @0 networkProxy)
-            benchmarksSeq
 
-        , bench_restoration @_ @ByronKey
+    let benchRestoreMultipleWallets nWallets = bench_restoration @_ @ShelleyKey
             networkProxy
             (trMessageText tr)
             socketFile
             np
             vData
-            (walletRnd "0-percent-rnd" $ mkRndAnyState @0)
-            benchmarksRnd
-
-        , bench_restoration @_ @ByronKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletRnd "0.1-percent-rnd" $ mkRndAnyState @1)
-            benchmarksRnd
-
-        , bench_restoration @_ @ByronKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletRnd "0.2-percent-rnd" $ mkRndAnyState @2)
-            benchmarksRnd
+            (map (\i -> walletSeq
+                ("w" <> T.pack (show i)) $ mkSeqAnyState' (Proxy @0) networkProxy)
+                [0..nWallets :: Int])
+            True -- Don't write progress to .timelog file(s)
+            (unsafeMkPercentage 0.01)
+            benchmarksSeq
 
-        , bench_restoration @_ @ByronKey
+    let benchRestoreRndWithOwnership permillProxy = bench_restoration
             networkProxy
             (trMessageText tr)
             socketFile
             np
             vData
-            (walletRnd "0.4-percent-rnd" $ mkRndAnyState @4)
+            [walletRnd (showPercent permillProxy <> "-percent-rnd")
+                $ mkRndAnyState' permillProxy networkProxy]
+            True -- Write progress to .timelog file
```

Comment:
Is this comment correct? The one in line 235 is opposite but the value is also True

Reply:
Thanks, the value was supposed to be `False` on line 235.

Could have added a new datatype to better represent this, but maybe I should more fundamentally refactor the `bench_restoration` function instead 🤔